Src/Sarif.PatternMatcher/ValidatorsCache.cs

```diff
@@ -228,6 +225,18 @@ public static void DisableValidationCaching(MethodInfo shouldCacheMethodInfo, bo
 
             if (validationMethods == null)
             {
+                if (context.TargetUri.IsAbsoluteUri)
```

Comment:
>context.TargetUri.IsAbsoluteUri [](http://example.com/codeflow?start=20&length=31)

this would only work if we run the tool in a machine. In our current scenario, the targeturi does not point to a physical location because it does not exist.

Maybe, we should not do this check and always apply the path hashing.

Reply:
ok! well, this helper requires a file path with a file.

what we can do next is to check and see whether we have the file contents available and just hash that instead, if it's there?

shall we check this in and then maybe do that follow on work? want to take that improvement? :)

You could also update the review sensitive data check to provide static validators and populate a *real* secret. for extra points, we could enable a test that ensures this plug-in also always includes a static validator for any plug in rule. :)